src/Compilers/Core/Portable/Diagnostic/SeverityFilter.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Represents a set of filtered diagnostic severities.
+    /// </summary>
+    internal readonly struct SeverityFilter
+    {
+        private const int HiddenBit = 0x01;
+        private const int InfoBit = 0x10;
+
+        private readonly int _flag;
+
+        private SeverityFilter(bool includeHidden, bool includeInfo)
```

Comment:
>bool includeHidden, bool includeInfo [](start = 31, length = 36)

Since this is a private constructor, consider passing in the bit value directly instead of two bools.